### ISSUE TITLE
Integration tests fail on local machine (but succeed in github.Actions)

### ISSUE THREAD

**Issue by Sheshuk** [SNEWS2/snewpy#370]
Hi snewpy developers! I've got a very strange problem, maybe someone could help me understand what is going on.

# In short
when I run the integration tests on my local machine, they fail (numbers are different from the cross-check table of expected neutrino interactions). 
The same tests succeed when run in CI/CD pipeline in github: see [here](https://github.com/SNEWS2/snewpy/actions/runs/12198655011/job/34187791418)

# Details

### What I did 
I make a fresh new installation of the latest snewpy main, into a fresh new virtual environment.

```shell
git clone -b main --single-branch git@github.com:SNEWS2/snewpy.git --depth=1 #faster, not downloading all the history
cd snewpy
python -m venv venv_test
source venv_test/bin/activate
python -m pip install ".[dev]"
pytest -m 'snowglobes'
```
### What I get

Discrepancy for `ar40k` detector. 
<details><summary>Test log</summary>
<p>

```
=================================================================================================================================================== test session starts ====================================================================================================================================================
platform linux -- Python 3.12.7, pytest-8.3.4, pluggy-1.5.0
rootdir: /home/ash/work/snews2/test/snewpy
configfile: pytest.ini
plugins: hypothesis-6.122.3
collected 92 items / 66 deselected / 26 selected                                                                                                                                                                                                                                                                           

python/snewpy/test/test_05_snowglobes.py FFFF                                                                                                                                                                                                                                                                        [ 15%]
python/snewpy/test/test_presn_rates.py ........                                                                                                                                                                                                                                                                      [ 46%]
python/snewpy/test/test_rate_calculation.py ..............                                                                                                                                                                                                                                                           [100%]

========================================================================================================================================================= FAILURES =========================================================================================================================================================
__________________________________________________________________________________________________________________________________ test_total_rate_equals_table_value[model_parameters0] ___________________________________________________________________________________________________________________________________

model_parameters = ('Bollig_2016', 's11.2c', 'AdiabaticMSW_IMO')

    @pytest.mark.parametrize('model_parameters',param_values)
    def test_total_rate_equals_table_value(model_parameters):
        fluence_file = fluence_calculation(*model_parameters)
        calculated_rates  = rates_calculation(fluence_file)
        for detector in detectors:
            expected = pytest.approx(rate_table[model_parameters][detector], rel=0.01)
>           assert calculated_rates[detector] == expected, f"Crosscheck failed for {detector}"
E           AssertionError: Crosscheck failed for ar40kt
E           assert np.float64(2682.988171379117) == 2509.8697955822163 ± 2.5e+01
E             
E             comparison failed
E             Obtained: 2682.988171379117
E             Expected: 2509.8697955822163 ± 2.5e+01

python/snewpy/test/test_05_snowglobes.py:41: AssertionError
--------------------------------------------------------------------------------------------------------------------------------------------------- Captured stdout call ---------------------------------------------------------------------------------------------------------------------------------------------------
/home/ash/.astropy/cache/snewpy/models/Bollig_2016/s11.2c
__________________________________________________________________________________________________________________________________ test_total_rate_equals_table_value[model_parameters1] ___________________________________________________________________________________________________________________________________

model_parameters = ('Bollig_2016', 's11.2c', 'AdiabaticMSW_NMO')

    @pytest.mark.parametrize('model_parameters',param_values)
    def test_total_rate_equals_table_value(model_parameters):
        fluence_file = fluence_calculation(*model_parameters)
        calculated_rates  = rates_calculation(fluence_file)
        for detector in detectors:
            expected = pytest.approx(rate_table[model_parameters][detector], rel=0.01)
>           assert calculated_rates[detector] == expected, f"Crosscheck failed for {detector}"
E           AssertionError: Crosscheck failed for ar40kt
E           assert np.float64(2862.1387086528507) == 2696.396378383758 ± 2.7e+01
E             
E             comparison failed
E             Obtained: 2862.1387086528507
E             Expected: 2696.396378383758 ± 2.7e+01

python/snewpy/test/test_05_snowglobes.py:41: AssertionError
--------------------------------------------------------------------------------------------------------------------------------------------------- Captured stdout call ---------------------------------------------------------------------------------------------------------------------------------------------------
/home/ash/.astropy/cache/snewpy/models/Bollig_2016/s11.2c
__________________________________________________________________________________________________________________________________ test_total_rate_equals_table_value[model_parameters2] ___________________________________________________________________________________________________________________________________

model_parameters = ('Bollig_2016', 's27.0c', 'AdiabaticMSW_IMO')

    @pytest.mark.parametrize('model_parameters',param_values)
    def test_total_rate_equals_table_value(model_parameters):
        fluence_file = fluence_calculation(*model_parameters)
        calculated_rates  = rates_calculation(fluence_file)
        for detector in detectors:
            expected = pytest.approx(rate_table[model_parameters][detector], rel=0.01)
>           assert calculated_rates[detector] == expected, f"Crosscheck failed for {detector}"
E           AssertionError: Crosscheck failed for ar40kt
E           assert np.float64(5469.603875098905) == 5192.001233928208 ± 5.2e+01
E             
E             comparison failed
E             Obtained: 5469.603875098905
E             Expected: 5192.001233928208 ± 5.2e+01

python/snewpy/test/test_05_snowglobes.py:41: AssertionError
--------------------------------------------------------------------------------------------------------------------------------------------------- Captured stdout call ---------------------------------------------------------------------------------------------------------------------------------------------------
/home/ash/.astropy/cache/snewpy/models/Bollig_2016/s27.0c
__________________________________________________________________________________________________________________________________ test_total_rate_equals_table_value[model_parameters3] ___________________________________________________________________________________________________________________________________

model_parameters = ('Bollig_2016', 's27.0c', 'AdiabaticMSW_NMO')

    @pytest.mark.parametrize('model_parameters',param_values)
    def test_total_rate_equals_table_value(model_parameters):
        fluence_file = fluence_calculation(*model_parameters)
        calculated_rates  = rates_calculation(fluence_file)
        for detector in detectors:
            expected = pytest.approx(rate_table[model_parameters][detector], rel=0.01)
>           assert calculated_rates[detector] == expected, f"Crosscheck failed for {detector}"
E           AssertionError: Crosscheck failed for ar40kt
E           assert np.float64(5789.6141397737465) == 5520.903869068218 ± 5.5e+01
E             
E             comparison failed
E             Obtained: 5789.6141397737465
E             Expected: 5520.903869068218 ± 5.5e+01

python/snewpy/test/test_05_snowglobes.py:41: AssertionError
--------------------------------------------------------------------------------------------------------------------------------------------------- Captured stdout call ---------------------------------------------------------------------------------------------------------------------------------------------------
/home/ash/.astropy/cache/snewpy/models/Bollig_2016/s27.0c
===================================================================================================================================================== warnings summary =====================================================================================================================================================
python/snewpy/_model_downloader.py:151
  /home/ash/work/snews2/test/snewpy/python/snewpy/_model_downloader.py:151: DeprecationWarning: open_text is deprecated. Use files() instead. Refer to https://importlib-resources.readthedocs.io/en/latest/using.html#migrating-from-legacy for migration advice.
    context = open_text("snewpy.models", "model_files.yml")

... omitted lots of warnings about missing efficiency etc ...

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
================================================================================================================================================= short test summary info ==================================================================================================================================================
FAILED python/snewpy/test/test_05_snowglobes.py::test_total_rate_equals_table_value[model_parameters0] - AssertionError: Crosscheck failed for ar40kt
FAILED python/snewpy/test/test_05_snowglobes.py::test_total_rate_equals_table_value[model_parameters1] - AssertionError: Crosscheck failed for ar40kt
FAILED python/snewpy/test/test_05_snowglobes.py::test_total_rate_equals_table_value[model_parameters2] - AssertionError: Crosscheck failed for ar40kt
FAILED python/snewpy/test/test_05_snowglobes.py::test_total_rate_equals_table_value[model_parameters3] - AssertionError: Crosscheck failed for ar40kt
=========================================================================================================================== 4 failed, 22 passed, 66 deselected, 539 warnings in 62.11s (0:01:02) ===========================================================================================================================
```

</p>
</details> 

## My environment
I'm using ArchLinux, Python 3.12.7
  
Current commit seems to be the same as in current main: 8fe22dd 
```shell
$ git log --oneline
8fe22dd (grafted, HEAD -> main, origin/main, origin/HEAD) Merge pull request #369 from SNEWS2/JostMigenda/ConsistentPrefixForGitTag
```

Packages versions:
```shell
$ pip freeze
astropy==7.0.0
astropy-iers-data==0.2024.12.9.0.36.21
attrs==24.2.0
certifi==2024.8.30
charset-normalizer==3.4.0
contourpy==1.3.1
cycler==0.12.1
fonttools==4.55.2
h5py==3.12.1
hypothesis==6.122.3
idna==3.10
iniconfig==2.0.0
kiwisolver==1.4.7
matplotlib==3.9.3
numpy==2.2.0
packaging==24.2
pandas==2.2.3
pillow==11.0.0
pluggy==1.5.0
pyerfa==2.0.1.5
pyparsing==3.2.0
pytest==8.3.4
python-dateutil==2.9.0.post0
pytz==2024.2
PyYAML==6.0.2
requests==2.32.3
scipy==1.14.1
six==1.17.0
snewpy @ file:///home/ash/work/snews2/test/snewpy
snowglobes_data==1.3.2
sortedcontainers==2.4.0
tqdm==4.67.1
tzdata==2024.2
urllib3==2.2.3
```

For comparison in the github [workflow](https://github.com/SNEWS2/snewpy/actions/runs/12198655011/job/34187791418), reading the part "Sucessfully installed..." I see the same versions:
```
Successfully installed 
astropy-7.0.0 
astropy-iers-data-0.2024.12.9.0.36.21 
attrs-24.2.0 
certifi-2024.8.30 
charset-normalizer-3.4.0 
contourpy-1.3.1 
cycler-0.12.1 
fonttools-4.55.2
h5py-3.12.1
hypothesis-6.122.3
idna-3.10
iniconfig-2.0.0
kiwisolver-1.4.7
matplotlib-3.9.3
numpy-2.2.0
packaging-24.2
pandas-2.2.3
pillow-11.0.0
pluggy-1.5.0
pyerfa-2.0.1.5
pyparsing-3.2.0
pytest-8.3.4
python-dateutil-2.9.0.post0
pytz-2024.2
pyyaml-6.0.2
requests-2.32.3
scipy-1.14.1
six-1.17.0
snewpy-1.6b1
snowglobes_data-1.3.2
sortedcontainers-2.4.0
tqdm-4.67.1
tzdata-2024.2
urllib3-2.2.3
```

## Summary
Discrepancy is only for `ar40k` detector, so apparently it's not a problem of SupernovaModel flux (otherwise it would fail on other detectors).
I thought this might be some problem with `snowglobes_data` package, but the versions are the same...

Any ideas where to look?

**Comment by JostMigenda** [comment]
Hi @Sheshuk, thanks for reporting this!

My mind immediately jumps to the fact that we had a fix for `ar40kt` in `snowglobes_data` v1.3.2, which changed the expected event rates. Looking at dd940fc15fc58927efe359540054632299d94f98, it seems that your event rates are almost exactly what we had before that change … 🤔
First idea: Can you check whether you have defined the `$SNOWGLOBES` environment variable to point to an older version of SNOwGLoBES? If so, that would override the `snowglobes_data` package …

**Comment by Sheshuk** [comment]
You're right, now I checked and it is set to a local snowglobes installation
(on both my computers at home and in office). Now the tests pass. Thanks a lot! One more mystery solved!

Probably we don't want this behavior anymore - as we're using `snowglobes_data`, we can drop the support of `$SNOWGLOBES` variable. If a user needs to use a custom snowglobes, they can use `pip install $SNOWGLOBES/snowglobes_data` to use their package in snewpy.
